apps/sms/js/thread_list_ui.js

```diff
@@ -903,28 +889,30 @@ var ThreadListUI = {
     if (!threadsContainer) {
       // We create the wrapper with a 'header' & 'ul'
       var threadsContainerWrapper =
-        ThreadListUI.createThreadContainer(timestamp);
+        this.createThreadContainer(timestamp);
       // Update threadsContainer with the new value
       threadsContainer = threadsContainerWrapper.childNodes[1];
       // Place our new content in the DOM
-      ThreadListUI.insertThreadContainer(threadsContainerWrapper, timestamp);
+      this.insertThreadContainer(threadsContainerWrapper, timestamp);
       // We had to create a container, so this will be the first thread in it.
       firstThreadInContainer = true;
     }
 
     // Where have I to place the new thread?
     var threads = threadsContainer.getElementsByTagName('li');
-    var threadFound = false;
-    for (var i = 0, l = threads.length; i < l; i++) {
-      if (timestamp > threads[i].dataset.time) {
-        threadFound = true;
-        threadsContainer.insertBefore(node, threads[i]);
+    var targetIdx = -1;
+    var lastIdx = threads.length - 1;
+    for (var i = lastIdx; i >= 0; i--) {
+      if (timestamp < threads[i].dataset.time) {
+        targetIdx = i;
         break;
       }
     }
 
-    if (!threadFound) {
+    if (targetIdx === lastIdx) {
       threadsContainer.appendChild(node);
+    } else {
+      threadsContainer.insertBefore(node, threads[targetIdx + 1]);
```

Comment:
I think you can just use inserBefore; if the second argument is null/undefined it's properly appended.